### ISSUE TITLE
Envoy Rate Limit should allow all actions to set the descriptor key 

### ISSUE THREAD

**Issue by medalliaerlich** [envoyproxy/envoy#5489]
*Title*: *Envoy Rate Limit should allow all actions to set the descriptor key *

*Description*:
When adding a Rate Limit action, for instance `generic_key`, one should be able to specify the descriptor's key. 
This example, one should be able to create two generic_keys and customize them:

- Generic key1:  descriptor_key: key1, descriptor_value: value1
- Generic key2:  descriptor_key: key2, descriptor_value: value2

In other words, this feature is already supported by the `header_values` action types, but not in the others.

This is actually a workaround for https://github.com/envoyproxy/envoy/issues/5521

**Comment by medalliaerlich** [comment]
Adding a comment to prevent from being closed.

**Comment by medalliaerlich** [comment]
Any news?
